virttest/env_process.py

```diff
@@ -314,6 +314,14 @@ def check_image(test, params, image_name, vm_process_status=None):
                               " Skip the image check.")
                 check_image_flag = False
 
+    # Save the potential bad image when the test is not passed.
+    # It should before image check.
+    if (params["test_passed"] == "False" and
+            params.get("save_image_on_error", "no") == "yes"):
+        if vm_process_status == "dead":
```

Comment:
Well it might be useful to backup even images when the vm is still alive. Yes, it might produce some corruptions, but it can still be useful.

Reply:
For the debug purpose, backup the image you mentioned does not useful.
As the next two comments, I don't understand why it need to `restore/backup/save` a "good" image when a test is finished.
It is not efficient for sure, and is it a necessary step?

Reply:
Well without powering the machine might still contain some useful data. Sure, some would be corrupt, some would be uncommitted (only in memory), but some data will be synced to disk already. I mean I don't insist on changing it, it just seems useful.

... maybe it'd be useful to at least log "Not creating bad backup, vm is running" to prevent people from trying to understand why they sometimes get backup and sometimes don't.

As for the backup, sure, it does not make sense. I incorrectly thought it's restoring the image instead of backuping (because src_good is `.backup` so I though the dst is `basename`) but it is not. The only reason to store the source as well I see is to allow double-checking what was the image like before testing (whether it was already corrupted, or whether it got corrupted in the test). But hopefully that should be identifiable from previous tests, so no problem with that.